### ISSUE TITLE
ErrorException Declaration of Illuminate\Database\Eloquent\Relations\Concerns\AsPivot::setKeysForSaveQuery(Illuminate\Database\Eloquent\Builder $query) should be compatible with Illuminate\Database\Eloquent\Model::setKeysForSaveQuery($query)

### ISSUE THREAD

**Issue by elfeffe** [laravel/framework#34585]
- Laravel Version: 8.7.0
- PHP Version: 7.4

@taylorotwell I see that you have fixed it in 2013 (https://github.com/laravel/framework/issues/2393), but I have updated it to L8 today and it changed the method as it was 7 years ago.
```
 /**
     * Set the keys for a save update query.
     *
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @return \Illuminate\Database\Eloquent\Builder
     */
    protected function setKeysForSaveQuery($query)
    {
        $query->where($this->getKeyName(), '=', $this->getKeyForSaveQuery());

        return $query;
    }
```
At 
`vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php
`

I have an Exception that says "ErrorException
Declaration of Illuminate\Database\Eloquent\Relations\Concerns\AsPivot::setKeysForSaveQuery(Illuminate\Database\Eloquent\Builder $query) should be compatible with Illuminate\Database\Eloquent\Model::setKeysForSaveQuery($query)"

It fixes injecting Builder to that method.

```
/**
     * Set the keys for a save update query.
     *
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @return \Illuminate\Database\Eloquent\Builder
     */
    protected function setKeysForSaveQuery(Builder $query)
    {
        $query->where($this->getKeyName(), '=', $this->getKeyForSaveQuery());

        return $query;
    }
```

Is it a bug?

**Comment by saleem-hadad** [comment]
Same here, any explanation?

**Comment by rodrigopedra** [comment]
It is fixed in release 8.7.1

**Comment by saleem-hadad** [comment]
I think this is fixed

**Comment by mattlibera** [comment]
yep, fixed (see #34578). `composer update` got me up to 8.7.1 and the problem is gone
